cli/src/test/java/keywhiz/cli/CliMainTest.java

```diff
@@ -20,30 +25,44 @@ public void renameCommandMapsToRenameActionConfig() {
   }
 
   @Test
-  public void parsesRenameCommandWithId() {
+  public void failsToParseRenameCommandWithoutResourceType() {
     CliMain.CommandLineParsingContext context = new CliMain.CommandLineParsingContext();
     JCommander commander = context.getCommander();
-    commander.parse("rename", "--id", "1", "newName");
 
-    String parsedCommand = commander.getParsedCommand();
-    assertEquals("rename", parsedCommand);
+    thrown.expect(ParameterException.class);
+    commander.parse("rename", "--oldName", "foo", "--newName", "bar");
+  }
 
-    RenameActionConfig config = (RenameActionConfig) context.getCommands().get(parsedCommand);
-    assertEquals(Long.valueOf(1L), config.secretId);
-    assertEquals("newName", config.newName);
+  @Test
+  public void failsToParseRenameCommandWithoutOldName() {
+    CliMain.CommandLineParsingContext context = new CliMain.CommandLineParsingContext();
+    JCommander commander = context.getCommander();
+
+    thrown.expect(ParameterException.class);
+    commander.parse("rename", "secret", "--newName", "bar");
+  }
+
+  @Test
+  public void failsToParseRenameCommandWithoutNewName() {
+    CliMain.CommandLineParsingContext context = new CliMain.CommandLineParsingContext();
+    JCommander commander = context.getCommander();
+
+    thrown.expect(ParameterException.class);
+    commander.parse("rename", "secret", "--oldName", "foo");
   }
```

Comment:
Now that our validation is at the JCommander level our missing-argument tests need to be as well.